tests/collatz-all-fail/expected_results.json

```diff
@@ -0,0 +1 @@
+{"version": 2, "status": "fail", "tests": [{"name": "zero steps for one", "status": "fail", "output": null, "test_code": "(test-success \"zero steps for one\" = collatz (quote (1)) 0)", "message": "Expected 0 but got (error = \"~s is not a number\" implement-me!)"}, {"name": "divide if even", "status": "fail", "output": null, "test_code": "(test-success \"divide if even\" = collatz (quote (16)) 4)", "message": "Expected 4 but got (error = \"~s is not a number\" implement-me!)"}, {"name": "even and odd steps", "status": "fail", "output": null, "test_code": "(test-success \"even and odd steps\" = collatz (quote (12)) 9)", "message": "Expected 9 but got (error = \"~s is not a number\" implement-me!)"}, {"name": "large number of even and odd steps", "status": "fail", "output": null, "test_code": "(test-success \"large number of even and odd steps\" = collatz (quote (1000000)) 152)", "message": "Expected 152 but got (error = \"~s is not a number\" implement-me!)"}]}
```

Comment:
Currently, the `test_code` also includes the function name:

```scheme
(test-success \"divide if even\" = collatz (quote (16)) 4)
```

However, we use the `name` property on the website and then display the `test_code` underneath it (C# example used):

![image](https://user-images.githubusercontent.com/135246/173005565-bc855dd7-bfda-4b50-82f9-333b5d5db142.png)

Maybe it would be better to remove the function definition and just have the assertion?

```scheme
collatz (quote (16)) 4
```

Reply:
It is now producing code that is equivalent to what the student would need to run locally to get the same "actual" output as a test.

Reply:
For clarity, the `test_code` for a given `test-success` is now:
``` scheme
(test-success _ comparator procedure (args ...)  expected) 
=> (comparator (procedure args ...) expected)
```
The `test_code` for a given `test-error` is now:
``` scheme
(test-error _ procedure (args ...))
=> (procedure args ...)
```